aws-sdk-core/lib/aws-sdk-core/param_validator.rb

```diff
@@ -60,7 +60,7 @@ def structure(ref, values, errors, context)
     def list(ref, values, errors, context)
       # ensure the value is an array
       unless values.is_a?(Array)
-        errors << "expected #{context} to be an array"
+        errors << "expected #{context} to be an array, got value #{values} (class: #{values.class}) instead."
```

Comment:
Also note, we should use the `#inspect` of the value for each error message instead of the default `#to_s` that results from passing the raw value through string interpolation. This makes for a better error message.